sentry_sdk/tracing_utils.py

```diff
@@ -872,10 +872,13 @@ def _generate_sample_rand(
 
     # Round down to exactly six decimal-digit precision.
     # Setting the context is needed to avoid an InvalidOperation exception
-    # in case the user has changed the default precision.
-    return Decimal(sample_rand).quantize(
-        Decimal("0.000001"), rounding=ROUND_DOWN, context=Context(prec=6)
-    )
+    # in case the user has changed the default precision or set traps.
+    with localcontext(DefaultContext) as ctx:
```

Comment:
Small readability nit, I would `import decimal` then change this to 

```suggestion
    with decimal.localcontext(DefaultContext) as ctx:
```

Otherwise, from just reading this code, it is unclear that the `localcontext` is a `decimal`-related thing